scylla-server/tests/nodes-services.test.ts

```diff
@@ -12,4 +37,17 @@ describe('Node', () => {
     // Use toEqual to compare parsedResult with the expected array
     expect(parsedResult).toEqual(expected);
   });
+
+  /**
+   * unit test for upsert node
+   * testing does nothing if node does exist
+   */
+  test('Upsert Node Does Nothing', async () => {
+    const expected = [{ name: 'test' }];
+    await upsertNode('test');
```

Comment:
if you want to test this you have to call it multiple times and make sure only one is in the database